testsuite/testsuite-cassandra/src/test/java/org/wildfly/swarm/cassandra/CassandraArquillianTest.java

```diff
@@ -0,0 +1,146 @@
+import org.wildfly.swarm.config.security.Flag;
+import org.wildfly.swarm.config.security.SecurityDomain;
+import org.wildfly.swarm.config.security.security_domain.ClassicAuthentication;
+import org.wildfly.swarm.config.security.security_domain.authentication.LoginModule;
+import org.wildfly.swarm.security.SecurityFraction;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Scott Marlow
+ */
+@RunWith(Arquillian.class)
+@DefaultDeployment
+public class CassandraArquillianTest {
+
+    @CreateSwarm
+    public static Swarm newSwarm() throws Exception {
+        return new Swarm(false)
+
+                .outboundSocketBinding("standard-sockets",
+                        new OutboundSocketBinding("casstesthost")
+                                .remoteHost("localhost")
+                                .remotePort(9042))
+                .fraction(SecurityFraction.defaultSecurityFraction()
+                        .securityDomain(
+                                new SecurityDomain("cassandraRealm")
+                                        .classicAuthentication(
+                                                new ClassicAuthentication().loginModule(
+                                                        new LoginModule("ConfiguredIdentity").code("ConfiguredIdentity")
+                                                                .flag(Flag.REQUIRED)
+                                                                .moduleOptions(new HashMap<Object, Object>() {
+                                                                                   {
+                                                                                       put("principal", "devuser");
+                                                                                       put("username", "devuser");
+                                                                                       put("password", "changethis");
+                                                                                   }
+                                                                               }
+                                                                )
+                                                )
+                                        )
+                        )
+                )
+                .fraction(new CassandraFraction()
+                .cassandra(new Cassandra("cassandratestprofile")
+                        .host(new Host("casstesthost")
+                            .outboundSocketBindingRef("casstesthost")
+                        )
+                        .jndiName("java:jboss/cassandradriver/test")
+                        .id("cassandratestprofile")
+                        .module("org.cassandra.custom")
+                        .securityDomain("cassandraRealm")
+                        .ssl(false)
+
+                )
+        );
```

Comment:
Since you config-api, then you should get a *lot* of configurable-via-YAML for free.

Run your stuff with -Dswarm.logging.org.wildfly.swarm.config=DEBUG and you'll see the dotted-notation version of things that can go into YAML.

To translate into YAML, view the dotted-notation as a tree.

`swarm.foo.bar=whatever`

becomes

    swarm:
      foo:
        bar: whatever